File: library/cc/request_trailers.cc
#include "request_trailers.h"

// NOLINT(namespace-envoy)

RequestTrailersBuilder RequestTrailers::to_request_trailers_builder() const {
  RequestTrailersBuilder builder;
  for (const auto& pair : this->all_headers()) {
    builder.set(pair.first, pair.second);
  }
  return builder;
}

File: library/kotlin/io/envoyproxy/envoymobile/RequestTrailers.kt
package io.envoyproxy.envoymobile

/*
 * Trailers representing an outbound request.
 */
@Suppress("EmptyClassBlock")
class RequestTrailers : Trailers {
  /**
   * Internal constructor used by builders.
   *
   * @param trailers: Trailers to set.
   */
  internal constructor(trailers: Map<String, List<String>>) : super(trailers)

  /**
   * Convert the trailers back to a builder for mutation.
   *
   * @return RequestTrailersBuilder, The new builder.
   */
  fun toRequestTrailersBuilder() = RequestTrailersBuilder(
    headers.mapValues {
      it.value.toMutableList()
    }.toMutableMap()
  )
}
